File: Exercism/fsharp/etl/Etl.fs
module Etl

let transform scoresWithLetters =
    scoresWithLetters
    |> Map.toSeq
    |> Seq.collect (fun (n, l) -> Seq.map (fun c -> (System.Char.ToLowerInvariant c, n)) l)
    |> Map.ofSeq

File: Exercism/csharp/etl/Etl.cs
using System.Collections.Generic;
using System.Linq;

public static class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old) => old
      .SelectMany(p => p.Value.Select(v => (v.ToLowerInvariant(), p.Key)))
      .ToDictionary(p => p.Item1, p => p.Item2);
}